### PR TITLE
Port code

### PR DESCRIPTION
**Pull request by pixelsoccupied**
- refactor for easier testing
- port and improve ztp repo's tool where it confirms exact match
- init dependencies 
- init github workflow

### CODE REVIEW

File: cmd/compare/compare.go [hunk 1]
```diff
@@ -1,37 +1,262 @@
...
+	if err != nil {
+		return nil
+	}
+
+	cmd.Flags().BoolVarP(&options.ExactMatchOnly, "exact-match-only", "", false, "Return early by determining if both sets are exact match")
+
+	return cmd
+}
+
+func (o compareOptions) validate() error {
```
Comment (imiller0) [inline cmd/compare/compare.go:69]: Good factoring. Allows test cases to be written against each functional method. Please add CI tests as you go along to ensure corner cases and basic functionality are preserved.

File: cmd/compare/compare.go [hunk 2]
```diff
@@ -1,37 +1,262 @@
...
+		os.Exit(1)
+	}
+
+	_ = k8sdiff.DiffProgram{
+		Exec:      nil,
+		IOStreams: genericclioptions.IOStreams{},
+	}
+}
+
+func getResourceFromPolicyIfAny(uList []unstructured.Unstructured) []unstructured.Unstructured {
```
Comment (imiller0) [inline cmd/compare/compare.go:119]: Especially for unstructure data please document what the expected format/content of the inputs/outputs are. This will significantly help when maintaining and bringing others into the project.

File: cmd/compare/compare.go [hunk 3]
```diff
@@ -1,37 +1,262 @@
...
 }
 
-func init() {
-	rootCmd.AddCommand(compareCmd)
+func yamlToUnstructured(file string) *unstructured.Unstructured {
+	yFile, _ := os.ReadFile(file)
+	newUnstructured := &unstructured.Unstructured{Object: map[string]interface{}{}}
+
+	if err := yaml.Unmarshal(yFile, &newUnstructured.Object); err != nil {
+		os.Exit(1)
```
Comment (imiller0) [inline cmd/compare/compare.go]: We should start working in failure mode error propagation from the start. At a minimum we need a reasonable error message here and should consider propagating the error up to the calling context.

Comment (pixelsoccupied) [reply cmd/compare/compare.go]: Updated to handle the error better + warning msg

File: cmd/compare/compare.go (hunk 2 above)
Comment (pixelsoccupied) [reply cmd/compare/compare.go:119]: Yep! Will be easier to understand once i start adding unit tests
